### ISSUE TITLE
🐛 HTML comments can't suppress diagnostics in embedded snippets

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#11841]
### Environment information

<details>

```bash

playground

```

</details>

### What happened?

https://biomejs.dev/playground/?lintRules=noDoubleEquals&tab=formatter&pane=Diagnostics&language=svelte#code=PAAhAC0ALQAgAGIAaQBvAG0AZQAtAGkAZwBuAG8AcgBlACAAbABpAG4AdAAvAHMAdQBzAHAAaQBjAGkAbwB1AHMALwBuAG8ARABvAHUAYgBsAGUARQBxAHUAYQBsAHMAOgAgAHIAZQBwAHIAbwBkAHUAYwB0AGkAbwBuACAALQAtAD4ACgA8AGQAaQB2ACAAZABhAHQAYQAtAGUAcQB1AGEAbAA9AHsAMQAgAD0APQAgACIAMQAiAH0APgA8AC8AZABpAHYAPgA%3D

```svelte
<!-- biome-ignore lint/suspicious/noDoubleEquals: reproduction -->
<div data-equal={1 == "1"}></div>
```

also: https://biomejs.dev/playground/?lintRules=noDoubleEquals&tab=formatter&pane=Diagnostics&language=svelte#code=PABkAGkAdgAgAAoAIAAgAC8ALwAgAGIAaQBvAG0AZQAtAGkAZwBuAG8AcgBlACAAbABpAG4AdAAvAHMAdQBzAHAAaQBjAGkAbwB1AHMALwBuAG8ARABvAHUAYgBsAGUARQBxAHUAYQBsAHMAOgAgAHIAZQBwAHIAbwBkAHUAYwB0AGkAbwBuAAoAIAAgAGQAYQB0AGEALQBlAHEAdQBhAGwAPQB7ADEAIAA9AD0AIAAiADEAIgB9AAoAIAAgAD4ACgA8AC8AZABpAHYAPgA%3D

```svelte
<div 
  // biome-ignore lint/suspicious/noDoubleEquals: reproduction
  data-equal={1 == "1"}
  >
</div>
```

### Expected result

The comment should suppress the diagnostic.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
